contracts/Liquidator/Liquidator.sol

```diff
@@ -305,6 +311,43 @@ contract Liquidator is Ownable2StepUpgradeable, ReentrancyGuardUpgradeable, Acce
         treasuryPercentMantissa = newTreasuryPercentMantissa;
     }
 
+    /**
+     * @notice Sets protocol share reserve contract address
+     * @param protocolShareReserve_ The address of the protocol share reserve contract
+     */
+    function setProtocolShareReserve(address payable protocolShareReserve_) external {
+        _checkAccessAllowed("setProtocolShareReserve(address)");
+        _setProtocolShareReserve(protocolShareReserve_);
+    }
+
+    /**
+     * @notice Reduce the reserves of the pending accumulated reserves
+     */
+    function reduceReserves() external {
+        _reduceReservesInternal();
+    }
+
+    function _reduceReservesInternal() internal {
+        for (uint256 index = pendingRedeem.length - 1; index > 0; index--) {
+            address vToken = pendingRedeem[index];
+            uint256 vTokenBalance = IVToken(vToken).balanceOf(address(this));
+            uint256 redeemResponse = IVBep20(vToken).redeem(vTokenBalance);
+            address underlying = IVBep20(vToken).underlying();
+
+            if (redeemResponse == 0) {
+                uint256 underlyingBalance = IVToken(underlying).balanceOf((address(this)));
```

Comment:
Just to avoid any confusion...
```suggestion
                uint256 underlyingBalance = IERC20Upgradeable(underlying).balanceOf((address(this)));
```